### PR TITLE
Metadata endpoint

### PR DESCRIPTION
**Pull request by ErlendHaa**

### CODE REVIEW

**Review by achaikou — COMMENTED** [review]
wow how fast my metadata wish came true even if it is long before Christmas :smile: 

Also - if we merge tests PR, what would you like to do with tests? Not sure how much value metadata query_tests would bring (in theory it is important endpoint, so you want similar tests), but 1 e2e is definitely good.
Would you want to include them here, or should I introduce these tests in later PR?

File: internal/vds/vds.cpp [hunk 1]
```diff
@@ -305,6 +305,20 @@ void set_voxels(
     }
 }
 
+void json_write_axis(
+    nlohmann::json& json,
+    int voxel_dim,
+    const OpenVDS::VolumeDataLayout *layout
+) {
+    json.push_back({
+        { "Annotation", layout->GetDimensionName(voxel_dim)       },
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply internal/vds/vds.cpp:314]: Ops

File: internal/vds/vds.cpp [hunk 2]
```diff
@@ -518,6 +518,36 @@ struct vdsbuffer fetch_fence(
...
+) {
+    OpenVDS::Error error;
+    OpenVDS::ScopedVDSHandle handle = OpenVDS::Open(vds, credentials, error);
+
+    if (error.code != 0) {
+        throw std::runtime_error("Could not open VDS: " + error.string);
+    }
+
+    auto access = OpenVDS::GetAccessManager(handle);
+    const auto *layout = access.GetVolumeDataLayout();
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply internal/vds/vds.cpp:533]: There is probably a lot that can be refactored in the cpp file. Gonna leave that for another PR

File: internal/vds/vds.cpp [hunk 3]
```diff
@@ -518,6 +518,36 @@ struct vdsbuffer fetch_fence(
...
+
+    nlohmann::json meta;
+    meta["format"] = layout->GetChannelFormat(0); //TODO turn into numpy-style format?
+    for (int i = 2; i >= 0 ; i--) {
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply internal/vds/vds.cpp:537]: I don't think we have immediate plans of supporting 4d seismic. I actually have a dimension check in fetch_slice. We might want to add that to all c++ functions

File: internal/vds/vds.cpp [hunk 4]
```diff
@@ -408,13 +422,7 @@ struct vdsbuffer fetch_slice_metadata(
...
     for (int i = 2; i >= 0; i--) {
         if (i == vdim) continue;
-        meta["axis"].push_back({
-            { "annotation", layout->GetDimensionName(i)       },
-            { "min",        layout->GetDimensionMin(i)        },
-            { "max",        layout->GetDimensionMax(i)        },
-            { "samples",    layout->GetDimensionNumSamples(i) },
-            { "unit",       layout->GetDimensionUnit(i)       },
-        });
+        json_write_axis(meta["axis"], i, layout);
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply internal/vds/vds.cpp:425]: In `TestMetadata` I marshal the buffer from c++ into the documentation Structs in go. Hence the test doubles as a safeguard that c++ and go structs are in sync

File: internal/vds/vds_test.go [hunk 5]
```diff
@@ -59,26 +61,9 @@ func TestMetadata(t *testing.T) {
...
 
-	if meta.Format != expected_format {
-		t.Fatalf(
-			"Expected format to be %v, was %v",
-			expected_format,
-			meta.Format,
-		)
-	}
+	if !reflect.DeepEqual(meta, expected) {
+		t.Fatalf("Expected %v, got  %v", expected, axis)
```
Comment (achaikou) [inline internal/vds/vds_test.go:66]: Not axis now, but meta.

File: internal/vds/vds_test.go [hunk 6]
```diff
@@ -42,11 +42,13 @@ func toFloat32(buf []byte) (*[]float32, error) {
 }
 
 func TestMetadata(t *testing.T) {
-	expected_format := 3
-	expected_axis := []Axis{
-		{ Annotation: "Inline",    Min: 1,  Max: 5,  Samples: 3, Unit: "unitless" },
-		{ Annotation: "Crossline", Min: 10, Max: 11, Samples: 2, Unit: "unitless" },
-		{ Annotation: "Sample",    Min: 4,  Max: 16, Samples: 4, Unit: "ms"       },
+	expected := Metadata{
+		Axis: []{
```
Comment (achaikou) [inline internal/vds/vds_test.go:46]: Also this doesn't build?

Comment (ErlendHaa) [reply internal/vds/vds_test.go]: I think I got a struct mid-implementation

Comment (ErlendHaa) [reply internal/vds/vds_test.go]: and another STROKE when writing the previous comment. Jeeez...

**Review by achaikou — APPROVED** [review]

**Pull request merged**